### PR TITLE
[luci-interpreter] Introduce Platform Abstraction Layer(PAL)

### PR DESCRIPTION
**Pull request by SlavikMIPT**
This commit introduces Platform Abstraction Layer(PAL)

ONE-DCO-1.0-Signed-off-by: Vyacheslav Bazhenov <slavikmipt@gmail.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/CMakeLists.txt [hunk 1]
```diff
@@ -1,31 +1,21 @@
...
-if (NOT TensorFlowEigenSource_FOUND)
-  message(STATUS "Skipping luci-interpreter: Eigen not found")
-  return()
-endif ()
-
-if (NOT TensorFlowRuySource_FOUND)
-  message(STATUS "Skipping luci-interpreter: Ruy not found")
-  return()
-endif ()
+message("LUCI INTERPRETER BEGIN")
```
Comment (seanshpark) [inline compiler/luci-interpreter/src/CMakeLists.txt:9]: Is `STAUTS` for `message` omitted on purpose?

Comment (SlavikMIPT) [reply compiler/luci-interpreter/src/CMakeLists.txt]: @seanshpark fixed

**Comment by SlavikMIPT** [comment]
@nnfw-bot test nncc-release

**Review by jinevening — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/luci-interpreter/pal/linux/pal.cmake [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    target_include_directories(${TGT} PRIVATE "${PAL}")
+    target_include_directories(${TGT} SYSTEM PRIVATE
+            "${TensorFlowRuySource_DIR}"
+            "${TensorFlowGEMMLowpSource_DIR}"
+            "${TensorFlowEigenSource_DIR}"
+            "${TensorFlowSource_DIR}")
+    target_include_directories(${TGT} PRIVATE ${LUCI_INTERPRETER_PAL_DIR})
+
+    # TODO put it back, I changed my mind.
+    # instead add sources with visitors in this library
```
Comment (binarman) [inline compiler/luci-interpreter/pal/linux/pal.cmake:42]: is this comment outdated?

File: compiler/luci-interpreter/pal/mcu/pal.cmake [hunk 3]
```diff
@@ -0,0 +1,49 @@
+macro(initialize_pal)
+    nnas_find_package(TensorFlowSource EXACT 2.3.0 QUIET)
+    nnas_find_package(TensorFlowGEMMLowpSource EXACT 2.3.0 QUIET)
```
Comment (binarman) [inline compiler/luci-interpreter/pal/mcu/pal.cmake:3]: I am not sure MCU mode needs this library, maybe add something like `#todo check if this is needed`?

File: compiler/luci-interpreter/pal/mcu/pal.cmake [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    # TODO put it back, I changed my mind.
+    # instead add sources with visitors in this library
+    set(PAL_SOURCES ${TensorFlowSource_DIR}/tensorflow/lite/kernels/internal/quantization_util.cc)
+    add_library(luci_interpreter_mcu_pal STATIC ${PAL_SOURCES})
+    set_target_properties(luci_interpreter_mcu_pal PROPERTIES POSITION_INDEPENDENT_CODE ON)
+    target_include_directories(luci_interpreter_mcu_pal SYSTEM PRIVATE "${TensorFlowSource_DIR}")
+
+    target_link_libraries(${TGT} PRIVATE luci_interpreter_mcu_pal)
+    #target_link_libraries(${TGT} PRIVATE Threads::Threads luci_interpreter_mcu_pal)
```
Comment (binarman) [inline compiler/luci-interpreter/pal/mcu/pal.cmake:48]: redundant line

File: compiler/luci-interpreter/pal/mcu/pal.cmake [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    target_include_directories(${TGT} PRIVATE "${PAL}")
+    target_include_directories(${TGT} SYSTEM PRIVATE
+            "${TensorFlowRuySource_DIR}"
+            "${TensorFlowGEMMLowpSource_DIR}"
+            "${TensorFlowEigenSource_DIR}"
+            "${TensorFlowSource_DIR}")
+    target_include_directories(${TGT} PRIVATE ${LUCI_INTERPRETER_PAL_DIR})
+
+    # TODO put it back, I changed my mind.
+    # instead add sources with visitors in this library
```
Comment (binarman) [inline compiler/luci-interpreter/pal/mcu/pal.cmake:41]: same as below

File: compiler/luci-interpreter/src/CMakeLists.txt [hunk 6]
```diff
@@ -1,31 +1,21 @@
...
-if (NOT TensorFlowEigenSource_FOUND)
-  message(STATUS "Skipping luci-interpreter: Eigen not found")
-  return()
-endif ()
-
-if (NOT TensorFlowRuySource_FOUND)
-  message(STATUS "Skipping luci-interpreter: Ruy not found")
-  return()
-endif ()
+message(STATUS "LUCI INTERPRETER BEGIN")
```
Comment (binarman) [inline compiler/luci-interpreter/src/CMakeLists.txt:9]: redundant messages?

**Review by binarman — APPROVED** [review]
LGTM, but fix comments (maybe in separate PR?)

File: compiler/luci-interpreter/src/CMakeLists.txt (hunk 6 above)
Comment (binarman) [reply compiler/luci-interpreter/src/CMakeLists.txt:9]: Maybe add more details, like "START CONFIGURING" instead of simple "BEGIN"?

**Pull request merged**